src/tools/run-moonbeam-fork.ts

```diff
@@ -74,17 +102,117 @@ const bootNodes = Object.values(NODE_KEYS)
 const main = async () => {
   // Variable to allow replaying some following steps if previous steps have been modified
   let hasChanged = false;
+  let polkadotVersion: string;
+
+  if (!argv.solo) {
+    const polkadotReleases = await (
+      await fetch("https://api.github.com/repos/paritytech/polkadot/releases")
+    ).json();
+
+    const latestPolkadotVersion = polkadotReleases.find((release) =>
+      release.assets.find((asset) => asset.name === "polkadot")
+    ).tag_name;
+
+    // Download new binary if non found, or if existing version doesnt match specified, or if not latest
```

Comment:
@crystalin  

This section is where the bulk of latest commit changes are. We now will download latest binary by default, or the version specified. If version downloaded is different to version specified (or if the latest is not the latest), it will replace that binary.